examples/fed-pubsub.cc

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2017-2019, Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC.  See the top-level NOTICE for additional details. All rights reserved.
+
+SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "helics/application_api/ValueFederate.hpp"
+#include "helics/application_api/Inputs.hpp"
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <regex>
+#include <set>
+#include <thread>
+#include <stdexcept>
+#include "helics/core/helicsCLI11.hpp"
+#include "helics/core/helicsVersion.hpp"
+
+int main (int argc, char *argv[])
+{
+    helics::helicsCLI11App app ("PubSub NS3 Example", "PubSubNS3Example");
+    std::string myname = "fed";
+    std::string publishKey = "ns3_test_value";
+
+    app.add_option ("--name,-n", myname, "name of this federate");
```

Comment:
I think for the purposes of the test, it should be okay get rid of the `myname` variable and name option, then just use the default federate name (`ns3`) created by the `SetupFederate` helper function.